src/field/arena.ts

```diff
@@ -824,57 +824,44 @@ export class Arena {
   }
 
   /**
-   * Attempts to get a tag from the Arena via {@linkcode getTagOnSide} that applies to both sides
-   * @param tagType The {@linkcode ArenaTagType} or {@linkcode ArenaTag} to get
-   * @returns either the {@linkcode ArenaTag}, or `undefined` if it isn't there
+   * Checks if an {@linkcode ArenaTag} exists on the specified side of the Arena.
+   * @param tagType - The {@linkcode ArenaTagType} to check for
+   * @param side - (Default `ArenaTagSide.BOTH`) The {@linkcode ArenaTagSide} to look at
+   * @returns Whether the `ArenaTag` exists
    */
-  getTag(tagType: ArenaTagType | AbstractConstructor<ArenaTag>): ArenaTag | undefined {
-    return this.getTagOnSide(tagType, ArenaTagSide.BOTH);
-  }
-
-  hasTag(tagType: ArenaTagType): boolean {
-    return !!this.getTag(tagType);
+  hasTag(tagType: ArenaTagType, side: ArenaTagSide = ArenaTagSide.BOTH): boolean {
+    return this.tags.some(
+      (t) => t.tagType === tagType && (side === ArenaTagSide.BOTH || t.side === ArenaTagSide.BOTH || t.side === side),
+    );
```

Comment:
NIT: I feel like the 2nd section of the `some()` condition is quite hard to understand if one doesn't know what's going on. It might be easier to pre-populate a `Set` of all the valid choices, or what do you think?

```suggestion
  hasTag(tagType: ArenaTagType, side: ArenaTagSide = ArenaTagSide.BOTH): boolean {
    const validSides= new Set<ArenaTagSide>([ArenaTagSide.BOTH, side]);
    
    return this.tags.some(
      (t) => t.tagType === tagType && validSides.has(t.side),
    );
```

Why a set? That way we don't have to worry about putting duplicates into it. The set will handle it for us